components/HistoryList/DesktopHistoryList/index.tsx

```diff
@@ -198,44 +198,65 @@ export default function DesktopHistoryList(props: Props) {
 
   React.useEffect(() => {
     let newTableData: LogResponse[] = deepCopy(props.logs)
+
+    if (props.internal) {
+      newTableData = newTableData.filter(
+        (log) => log.item.itemDefinition.internal
+      )
+    }
+
     if (props.search) {
       const search = props.search.toLowerCase()
-      newTableData = [
-        ...newTableData.filter((log) => {
-          return (
-            log.staff.name.toLowerCase().includes(search) ||
-            log.item.itemDefinition.name.toLowerCase().includes(search) ||
-            (log.item.attributes &&
-              log.item.attributes
-                .map((attr) =>
-                  `${attr.attribute.name}: ${attr.value}`.toLowerCase()
-                )
-                .join(' ')
-                .includes(search)) ||
-            (log.item.itemDefinition.category &&
-              log.item.itemDefinition.category.name
-                .toLowerCase()
-                .includes(search)) ||
-            log.quantityDelta.toString().toLowerCase().includes(search) ||
-            DateToReadableDateString(log.date).toLowerCase().includes(search)
-          )
-        }),
-      ]
+      newTableData = newTableData.filter((log) => {
+        return (
+          log.staff.name.toLowerCase().includes(search) ||
+          log.item.itemDefinition.name.toLowerCase().includes(search) ||
+          (log.item.attributes &&
+            log.item.attributes
+              .map((attr) =>
+                `${attr.attribute.name}: ${attr.value}`.toLowerCase()
+              )
+              .join(' ')
+              .includes(search)) ||
+          (log.item.itemDefinition.category &&
+            log.item.itemDefinition.category.name
+              .toLowerCase()
+              .includes(search)) ||
+          log.quantityDelta.toString().toLowerCase().includes(search) ||
+          DateToReadableDateString(log.date).toLowerCase().includes(search)
+        )
+      })
+    }
+
+    if (props.startDate || props.endDate) {
```

Comment:
^ would be better to just split this so we don't use random dates...

Reply:
Discussed this with Zavier offline, I used the far away dates for the sake of brevity, and he agreed. It's a bit of a moot point, though, so I'll go ahead and make the change in CCAHT-161 #95, since I move the filtering out of the component and into the page in that PR.